src/test/scala/mesosphere/marathon/tasks/TaskTrackerTest.scala

```diff
@@ -474,4 +477,47 @@ class TaskTrackerTest extends MarathonSpec with Matchers {
 
     verify(state, times(1)).update(any())
   }
+
+  // sounds strange, but this is how it currently works: checkStagedTasks will only consider a missing startedAt to
+  // determine whether a task is in staging and might need to be killed if it exceeded the taskLaunchTimeout
+  test("ensure that checkStagedTasks returns tasks disregarding the stagedAt property") {
+    import scala.language.implicitConversions
+    implicit def toMillis(timestamp: Timestamp): Long = timestamp.toDateTime.getMillis
+
+    val clock = ConstantClock(Timestamp.now())
+    val now = clock.now()
+
+    val unstagedTask = MarathonTask.newBuilder()
+      .setId("unstaged")
+      .build()
+
+    val stagedTask = MarathonTask.newBuilder()
+      .setId("staged")
+      .setStagedAt(now - 10.seconds)
+      .build()
+
+    val withoutStagedAtTask = MarathonTask.newBuilder()
+      .setId("withoutStagedAt")
+      .setStagedAt(now - 10.days)
+      .build()
+
+    val runningTask = MarathonTask.newBuilder()
+      .setId("running")
+      .setStartedAt(now - 5.seconds)
+      .build()
+
+    // Given("several tasks of which not all have been staged yet")
+    taskTracker.created(TEST_APP_NAME, unstagedTask)
+    taskTracker.created(TEST_APP_NAME, withoutStagedAtTask)
+    taskTracker.created(TEST_APP_NAME, stagedTask)
+    taskTracker.created(TEST_APP_NAME, runningTask)
+
+    // When("staged tasks are queried")
+    val stagedTasks = taskTracker.checkStagedTasks(now)
+
+    // Then("Only the stagedTask ")
```

Comment:
I didn't want to mixin the GivenWhenThen trait into a test that otherwise doesn't make use of it, and I didn't want to also change the remaining. Mixed it in now, but only used it in the added test.